### ISSUE TITLE
Meson+Ninja can't compile OpenMSX on NixOS

### ISSUE THREAD

**Issue by AndersonTorres** [openMSX/openMSX#1370]
Hello! I am Anderson Torres, the maintainer of openmsx package for Nixpkgs.

Trying the new Meson building system, I have found a linking error. Below is the full log:

https://hastebin.com/uxaxiginas.apache

Last lines (some lines are **huge**):

```
openmsx> /nix/store/hy3lz2vfv9qq2v5jz9nzlx6mmiaq79rj-binutils-2.35.1/bin/ld: openmsx.p/src_DeviceFactory.cc.o: in function `openmsx::DeviceFactory::create(openmsx::DeviceConfig const&)':
openmsx> DeviceFactory.cc:(.text+0x1636): undefined reference to `openmsx::MSXToshibaTcx200x::MSXToshibaTcx200x(openmsx::DeviceConfig const&)'
openmsx> collect2: error: ld returned 1 exit status
openmsx> [553/572] Compiling C++ object unittest.p/src_unittest_Math_test.cc.or_test.cc.o
openmsx> ninja: build stopped: subcommand failed.
```

On the other hand, the default, custom build system compiles it OK - at least according to OfBorg:

https://logs.nix.ci/?attempt_id=eb8c599f-e999-476a-9594-08a333ee3a96&key=nixos%2Fnixpkgs.132280

**Comment by mthuurne** [comment]
The problem seems to be that we simply forgot to add a new source file to the Meson sources list. I pushed a fix: f4d74e3761b5c55fcaa1242a42a54b70790ddcfc, although that also adds `TalentTDC600` which was implemented after the 17.0 release. You might want to pick 6de42a4fbe9495cb5ee0301d8c8693d79986759f as well: it wouldn't build for me with the latest Meson without that patch.

Note that the Meson build for openMSX is a proof of concept, not ready for production use. It's good that you're testing it, but for publishing packages, please use the custom build system instead.

**Comment by AndersonTorres** [comment]
Okay, then. That bein said, I will close this issue.
